### PR TITLE
Documentation: add a CLI reference section

### PR DESCRIPTION
**Pull request by aalemayhu**
Generated via Cobra. Added it to the install file after the agent command reference.

I saw the below warnings from `vndr`, do we need to add them as well?

```
  2017/11/17 09:47:27 WARNING: dependency is not vendored: github.com/cpuguy83/go-md2man/md2man
2017/11/17 09:47:27 WARNING: dependency is not vendored: github.com/russross/blackfriday
```

### CODE REVIEW

**Comment by aalemayhu** [comment]
Looks like those dependencies are needed for the build. Will monitor the build to see if something else is missing.

**Review by joestringer — APPROVED** [review]
Looks neat. From a review perspective and future reference, it's a little easier to review if the autogenerated files are added in an entirely separate commit from the other changes, so we can put more focus into ensuring the handwritten code is correct and just scan across the autogenerated code to check that it looks reasonable. Similarly for vendor/, since a tool generates that as well.

LGTM to me from CLI perspective.

This PR adds a test to `tests/`, please consider filing a ticket to move this over to Ginkgo when we can.

**Comment by aalemayhu** [comment]
For the generated files in docs dir a new commit is fine, but the vendor. I would prefer committing dependencies together with what they were used to avoid any new warnings introduced. You can still use `git` to ignore all these files.

**Comment by joestringer** [comment]
Fair enough. My workflow typically involves writing the piece that will use the new dependency, running the vendor tool to grab the dependency, then committing the dependency by itself, then committing the code that uses it. But I can understand if you'd prefer not to do the same.

**Comment by aalemayhu** [comment]
Hmm, this is still failing because when the monitor command file is being created the order of the events is still changing. I thought using `sort.Strings` like below would resolve this. Will try not using append and see how that works 

```go
func listEventTypes() []string {
	types := []string{}
	for k := range eventTypes {
		types = append(types, k)
	}
	sort.Strings(types)
	return types
}
```

Failure

```
07:00:51 ==> cilium-master-build-PR-2079-4: ----- 00-check-cmdref.sh: main: Detected a difference in the cmdref directory
07:00:51 ==> cilium-master-build-PR-2079-4: ----- 00-check-cmdref.sh: main: diff --git a/Documentation/cmdref/cilium_monitor.md b/Documentation/cmdref/cilium_monitor.md
07:00:51 ==> cilium-master-build-PR-2079-4: index 076476b..b04b95b 100644
07:00:51 ==> cilium-master-build-PR-2079-4: --- a/Documentation/cmdref/cilium_monitor.md
07:00:51 ==> cilium-master-build-PR-2079-4: +++ b/Documentation/cmdref/cilium_monitor.md
07:00:51 ==> cilium-master-build-PR-2079-4: @@ -23,7 +23,7 @@ cilium monitor
07:00:51 ==> cilium-master-build-PR-2079-4:        --hex                 Do not dissect, print payload in HEX
07:00:51 ==> cilium-master-build-PR-2079-4:        --related-to uint16   Filter by either source or destination endpoint id
07:00:51 ==> cilium-master-build-PR-2079-4:        --to uint16           Filter by destination endpoint id
07:00:51 ==> cilium-master-build-PR-2079-4: -  -t, --type string         Filter by event types [drop debug capture trace]
07:00:51 ==> cilium-master-build-PR-2079-4: +  -t, --type string         Filter by event types [capture debug drop trace]
07:00:51 ==> cilium-master-build-PR-2079-4:    -v, --verbose             Enable verbose output
07:00:51 ==> cilium-master-build-PR-2079-4:  ```
07:00:51 ==> cilium-master-build-PR-2079-4:  
07:00:51 ==> cilium-master-build-PR-2079-4: ------------------------------------------------------------------------
07:00:51 ==> cilium-master-build-PR-2079-4:                             Test Failed
07:00:51 ==> cilium-master-build-PR-2079-4: Please rerun 'make -C Documentation cmdref' and commit your changes
07:00:51 ==> cilium-master-build-PR-2079-4: 
07:00:51 ==> cilium-master-build-PR-2079-4: ------------------------------------------------------------------------
```

**Review by aanm — APPROVED** [review]

File: cilium/cmd/cmdref.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+)
+
+var dir string
+
+var cmdRef = &cobra.Command{
+	Use:   "cmdref",
+	Short: "Generate Cilium command reference",
+	Run: func(cmd *cobra.Command, args []string) {
```
Comment (tgraf) [inline cilium/cmd/cmdref.go:30]: Set `Hidden: true`?
https://godoc.org/github.com/spf13/cobra#Command

File: tests/00-check-cmdref.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
```
Comment (tgraf) [inline tests/00-check-cmdref.sh]: Can we run this as part of the build instead or is it too expensive?

**Review by tgraf — CHANGES_REQUESTED** [review]

File: tests/00-check-cmdref.sh (hunk 2 above)
Comment (aalemayhu) [reply tests/00-check-cmdref.sh]: We can do that, but I still, want the runtime check which can prevent someone from forgetting to check-in files.

File: cilium/cmd/cmdref.go (hunk 1 above)
Comment (aalemayhu) [reply cilium/cmd/cmdref.go:30]: Cool was not aware of that. Makes sense to hide it.

File: tests/00-check-cmdref.sh (hunk 2 above)
Comment (tgraf) [reply tests/00-check-cmdref.sh]: OK, let's do both then

**Review by tgraf — APPROVED** [review]

**Pull request merged**